doc/building.md

```diff
@@ -89,9 +89,7 @@ on where and how to check out the source code.
         directory. This is especially important if your user name contains
         spaces and/or mixed upper and lower case letters.
 
-      * Clone the JDK repository using the Cygwin command line `hg` client
-        as instructed in this document. That is, do *not* use another Mercurial
-        client such as TortoiseHg.
+      * Clone the JDK repository using [Git for Windows](https://gitforwindows.org).
```

Comment:
I think I said before, this is *not* a recommended way! Instead, git from Cygwin is still recommended.

Git for Windows will for instance set the coreautolf flag. Now we have pushed a work-around for this, but I object to *recommending* git for windows.

Reply:
BTW the value of coreautolf flag is asked during the installation of Git for windows, it is should be mentioned in the docs, since it is quite common issue.

Reply:
@jddarcy Recommended new wording:
```
      * Clone the JDK repository using the Cygwin command line `git` client
        as instructed in this document.

        While using the Cygwin `git` client is the recommended way, since it
        guarantees that there will be no line ending issues, and it works well
        with other Cygwin tools, it is also possible to use [Git for
        Windows](https://gitforwindows.org). If you chose this path, make sure
        to set `core.autocrlf` to `false` (this is asked during installation).
```

Reply:
I would recommend putting in a link to the instructions on the Skara wiki page. They actually recommend Git for Windows and I have switched too. The main reason is to be able to use the token manager. To just clone and build, using Cygwin Git is definitely easier though.

Reply:
@erikj79 

Ok, here is a completely rewritten portion, which highlights the differences between Cygwin git and Git for Windows.

```suggestion
      * You need to install a git client. You have two choices, Cygwin git or
        Git for Windows. Unfortunately there are pros and cons with each choice.

        * The Cygwin `git` client has no line ending issues and understands
          Cygwin paths (which are used throughout the JDK build system).
          However, it does not currently work well with the Skara CLI tooling.
          Please see the [Skara wiki on Git clients](
          https://wiki.openjdk.java.net/display/SKARA/Skara#Skara-Git) for
          up-to-date information about the Skara git client support.

        * The [Git for Windows](https://gitforwindows.org) client has issues
          with line endings, and do not understand Cygwin paths. It does work
          well with the Skara CLI tooling, however. To alleviate the line ending
          problems, make sure you set `core.autocrlf` to `false` (this is asked
          during installation).
```

@jddarcy Do you find this acceptable? This PR has been open for almost a month, and it's starting getting embarrassing that our build instruction does not match the new Github reality. :(

Reply:
You should add a `.gitattributes` file which defines the desired line endings for each file type or for all files. Recommending a specific git client just because of that (?) is unnecessary.

Here is a arbitrary example:
https://github.com/alexkaratarakis/gitattributes/blob/master/Java.gitattributes

Reply:
test

Reply:
I like your proposed text, Magnus.